mmhuman3d/core/visualization/renderer/torch3d_renderer/pointcloud_renderer.py

```diff
@@ -52,84 +40,73 @@ def __init__(self,
+            out_img_format=out_img_format,
+            **kwargs)
+
+    def to(self, device):
+        if isinstance(device, str):
+            device = torch.device(device)
+        self.device = device
+        if getattr(self.rasterizer, 'cameras', None) is not None:
+            self.rasterizer.cameras = self.rasterizer.cameras.to(device)
 
-    def set_render_params(self, **kwargs):
+        self.compositor = self.compositor.to(device)
+        return self
+
+    def _init_renderer(self, rasterizer=None, compositor=None, **kwargs):
         """Set render params."""
-        self.raster_type = kwargs.get('raster_type', 'point')
-        self.shader_type = None
-        self.bg_color = torch.tensor(
-            kwargs.get('bg_color', [
-                1.0,
-                1.0,
-                1.0,
-                0.0,
-            ]),
-            dtype=torch.float32,
-            device=self.device)
-        self.raster_settings = PointsRasterizationSettings(
-            image_size=self.resolution,
-            radius=self.radius
-            if self.radius is not None else kwargs.get('radius'),
-            points_per_pixel=kwargs.get('points_per_pixel', 10))
-
-    def init_renderer(self, cameras):
-        renderer = PointsRenderer(
-            rasterizer=build_raster(
-                dict(
-                    type=self.raster_type,
-                    cameras=cameras,
-                    raster_settings=self.raster_settings)),
-            compositor=AlphaCompositor(background_color=self.bg_color))
-        return renderer
+
+        if isinstance(rasterizer, nn.Module):
+            rasterizer.raster_settings.image_size = self.resolution
+            self.rasterizer = rasterizer
+        elif isinstance(rasterizer, dict):
+            rasterizer['image_size'] = self.resolution
+            if self.radius is not None:
+                rasterizer.update(radius=self.radius)
+            raster_settings = PointsRasterizationSettings(**rasterizer)
+            self.rasterizer = PointsRasterizer(raster_settings=raster_settings)
+        elif rasterizer is None:
+            self.rasterizer = PointsRasterizer(
+                raster_settings=PointsRasterizationSettings(
+                    radius=self.radius,
+                    image_size=self.resolution,
+                    points_per_pixel=10))
+        else:
+            raise TypeError(
+                f'Wrong type of rasterizer: {type(self.rasterizer)}.')
+
+        if isinstance(compositor, dict):
+            self.compositor = AlphaCompositor(**compositor)
+        elif isinstance(compositor, nn.Module):
+            self.compositor = compositor
+        elif compositor is None:
+            self.compositor = AlphaCompositor()
+        else:
+            raise TypeError(
+                f'Wrong type of compositor: {type(self.compositor)}.')
+        self = self.to(self.device)
 
     def forward(
         self,
         pointclouds: Optional[Pointclouds] = None,
```

Comment:
`pointclouds` cannot be None according to the `pointclouds = pointclouds.to(self.device)`

Reply:
ok